modules/xphoto/src/learning_based_color_balance.cpp

```diff
@@ -461,7 +461,8 @@ void extractSimpleFeatures(InputArray _src, OutputArray _dst, int range_max_val,
 inline float regressionTreePredict(Vec2f src, uchar *tree_feature_idx, float *tree_thresh_vals, float *tree_leaf_vals)
 {
     int node_idx = 0;
-    int depth = (int)round(log(num_tree_nodes) / log(2));
+	//result always positive, so we can replace round as static_cast + .5
+    int depth = static_cast<int>( (log(static_cast<float>(num_tree_nodes)) / log(static_cast<float>(2))) + .5);
```

Comment:
Sure, cvRound should work as well. I would also replace "static_cast<float>(2)" with simply "2.0f".